### PR TITLE
Document publishing snapshots and releases to different repositories

### LINKED ISSUE #5095: Document how to publish snapshots and releases in a different repositories using the maven-publish plugin

### PR DESCRIPTION
**Pull request by marcphilipp**
Resolves #5095.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/publishingMaven.adoc [hunk 1]
```diff
@@ -175,13 +175,32 @@ If a project defines multiple publications then Gradle will publish each of thes
...
     <sourcefile file="build.gradle" snippet="repositories"/>
 </sample>
 ++++
 
 The DSL used to declare repositories for publication is the same DSL that is used to declare repositories to consume dependencies from, api:org.gradle.api.artifacts.dsl.RepositoryHandler[]. However, in the context of Maven publication only api:org.gradle.api.artifacts.repositories.MavenArtifactRepository[] repositories can be used for publication.
 
+[[publishing_maven:snapshot_and_release_repositories]]
+==== Snapshot and release repositories
+
+It is a common practice to publish snapshots and releases to different Maven repositories. A simple way to accomplish this is to configure the repository URL based on the project version.
```
Comment (pledbrook) [inline subprojects/docs/src/docs/userguide/publishingMaven.adoc]: When you're leading into an example, end the paragraph with a colon. Alternatively, explicitly mention what you're demonstrating in the sample that follows the paragraph.

For example:

    A simple way to accomplish this is to configure the repository URL based on the project version. The following sample uses one URL for versions that end with "-SNAPSHOT" and a different URL for the rest:

BTW, I got lost amongst all the START/END SNIPPET lines. I assume interleaving those works. Is that correct?

File: subprojects/docs/src/docs/userguide/publishingMaven.adoc [hunk 2]
```diff
@@ -175,13 +175,32 @@ If a project defines multiple publications then Gradle will publish each of thes
...
+
+++++
+<sample dir="maven-publish/javaProject" id="publishing_maven:example:repo-url-from-variable" title="Configuring repository URL based on project version">
+    <sourcefile file="build.gradle" snippet="repo-url-from-variable"/>
+</sample>
+++++
+
+Similarly, you can use a <<build_environment, project or system property>> to decide for one repository or another. The following example uses the release repository if the project property `release` is set, e.g. by running `gradle -Prelease publish`.
```
Comment (pledbrook) [inline subprojects/docs/src/docs/userguide/publishingMaven.adoc]: to decide which repository to publish to.

or

    to decide on one repository over another

I think it's also important to clarify what the e.g. refers to. For example:

    The following example/sample uses the release repository if the project property `release` is set, such as when a user runs `gradle -Prelease publish`:

This just seems a little clearer to me.

File: subprojects/docs/src/samples/maven-publish/javaProject/build.gradle [hunk 3]
```diff
@@ -27,14 +27,28 @@ publishing {
             }
         }
     }
+}
 // END SNIPPET publish-custom-artifact
+
+// START SNIPPET repo-url-from-variable
+ext {
```
Comment (oehme) [inline subprojects/docs/src/samples/maven-publish/javaProject/build.gradle]: Why put this in the ext namespace? A local variable in the publishing block would be simpler.

**Comment by marcphilipp** [comment]
I applied the proposed changes in the latest update of this PR.

@pledbrook PTAL

File: subprojects/docs/src/docs/userguide/publishingMaven.adoc [hunk 4]
```diff
@@ -175,13 +175,32 @@ If a project defines multiple publications then Gradle will publish each of thes
...
     <sourcefile file="build.gradle" snippet="repositories"/>
 </sample>
 ++++
 
 The DSL used to declare repositories for publication is the same DSL that is used to declare repositories to consume dependencies from, api:org.gradle.api.artifacts.dsl.RepositoryHandler[]. However, in the context of Maven publication only api:org.gradle.api.artifacts.repositories.MavenArtifactRepository[] repositories can be used for publication.
 
+[[publishing_maven:snapshot_and_release_repositories]]
+==== Snapshot and release repositories
+
+It is a common practice to publish snapshots and releases to different Maven repositories. A simple way to accomplish this is to configure the repository URL based on the project version. The following sample uses one URL for versions that end with “`SNAPSHOT`” and a different URL for the rest:
```
Comment (pledbrook) [inline subprojects/docs/src/docs/userguide/publishingMaven.adoc]: “`SNAPSHOT`” <- There's no reason to make this monospace, particularly with quotes. Also, please use normal quotes, not smart ones. Try: "SNAPSHOT"

Comment (marcphilipp) [reply subprojects/docs/src/docs/userguide/publishingMaven.adoc]: Will do! Out of curiosity: Why not smart quotes?

**Comment by marcphilipp** [comment]
@pledbrook One more time, please. 🙂

File: subprojects/docs/src/docs/userguide/publishingMaven.adoc (hunk 4 above)
Comment (pledbrook) [reply subprojects/docs/src/docs/userguide/publishingMaven.adoc]: Good question. I will justify my stance with: consistency :smile:

If I'm honest, I'm not a fan of curly quotes. But I also think it's awkward to use curly quotes in prose and straight quotes in code when you're using the same editor for both.

**Pull request merged**